wsd/DocumentBroker.cpp

```diff
@@ -389,6 +389,12 @@ void DocumentBroker::pollThread()
             continue;
         }
 
+        if (isModified() && _storage->queryLocalFileToStorageAsyncUploadState().state() == StorageBase::AsyncUpload::State::Running)
+        {
+            _stop = false;
```

Comment:
Can't we add this check before setting `_stop = true;` ?

We should be able to do that when we get the socket disconnection, or whenever the trigger condition is met.

Resetting the _stop flag has many obvious issues, but one thing that might not be obvious is that some other logic might have already been triggered by the time we reset it, but we can't undo that logic (certainly not when we cannot know about it here). 

So it's always best to avoid changing the _stop flag in the first place.